### ISSUE TITLE
Foreign Keys 'no such table'

### ISSUE THREAD

**Issue by Nailik** [simolus3/moor#454]
Hello,

i try to use foreign keys in my tables but it seems like the table names in the custom constraint get replaced.

The Column in my Table looks as following:
` TextColumn get artCrc => text().nullable().customConstraint('NULL REFERENCES arts(crc)')();`

When i try to insert a row it get:
`Unhandled Exception: SqliteException: no such table: main.Art, SQL logic error`

This seems logically because there simply is no table named `Art` it's name is `arts`. Also when i check my database in `DB Browser for SQLite` i see in the `CREATE TABLE` statement 
`[...] art_crc VARCHAR NULL REFERENCES Art(crc) [...]`.

Did i write sth wrong or is there a bug in moor that replaces table names?

**Comment by simolus3** [comment]
This seems weird, moor will not rewrite any `customConstraint`. Can you try to re-run the builder and enable `logStatements: true` on the `VmDatabase`? This will make moor print the statements its running, so we can see the `CREATE TABLE` statement it issues.

Another thing to be aware of is that you'll need to make sure the tables are created in the right order, but it seems like there is another problem here.

**Comment by Nailik** [comment]
I think the problem was really the sorting.
I know sorted correctly and in the statements i get `[...] art_crc VARCHAR NULL REFERENCES arts(crc) [...]` now.
But even after wrong arranging the tables i cannot reproduce the problem before.
I tried: uninstalling, removing `moor_database.g.dart`, `flutter clean`, `flutter pub get`, `flutter packages pub run build_runner watch` and reinstalling - didn't get the error anymore.

So in conclusion probably the issue was wrong otder of the tables in `@UseMoor(tables:`
